src/metabase/lib/breakout.cljc

```diff
@@ -113,7 +113,9 @@
           (filter (fn [[_ref {:keys [join-alias source-field]} _id-or-name :as a-breakout]]
                     (and (lib.equality/find-matching-column query stage-number a-breakout [column] {:generous? true})
                          (= source-field (:fk-field-id column)) ; Must match, including both being nil/missing.
-                         (= join-alias   (::lib.join/join-alias column))  ; Must match, including both being nil/missing.
+                         (or
```

Comment:
@ericnormand can this be removed at all? I thought that `lib.equality` handles join mismatches already.

Reply:
What do you want to remove?

Reply:
`join-alias` checks

Reply:
I can try it and run the tests. I think it's beyond me to understand the consequences.

Reply:
Ultimately, the kind of check that you added is not a way to go. We should not destruct field refs like this and rely on some specific `alias`